arcgis-ios-sdk-samples/Layers/Edit KML ground overlay/EditKMLGroundOverlayViewController.swift

```diff
@@ -0,0 +1,79 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class EditKMLGroundOverlayViewController: UIViewController {
+    // MARK: Storyboard views and properties
+    @IBOutlet var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene(groundOverlay: overlay)
+        }
+    }
+    // The slider that controls the overlay's opacity.
+    @IBOutlet var opacitySlider: UISlider!
+    // The label that displays the slider's value.
+    @IBOutlet var valueLabel: UILabel!
+    
+    // The KML ground overlay.
+    let overlay: AGSKMLGroundOverlay = {
+        // Create a geometry for the ground overlay.
+        let overlayGeometry = AGSEnvelope(xMin: -123.066227926904, yMin: 44.04736963555683, xMax: -123.0796942287304, yMax: 44.03878298600624, spatialReference: .wgs84())
+        // Create a KML icon for the overlay image.
+        let imageURL = URL(string: "https://libapps.s3.amazonaws.com/accounts/55937/images/1944.jpg")!
+        let overlayImage = AGSKMLIcon(url: imageURL)
+        return AGSKMLGroundOverlay(geometry: overlayGeometry, icon: overlayImage)!
+    }()
+    
+    // MARK: Actions and methods
+    @IBAction func sliderValueChanged(_ slider: UISlider) {
+        // Change the color of the overlay according to the slider's value.
+        let alpha = CGFloat(slider.value)
+        overlay.color = UIColor.black.withAlphaComponent(alpha)
+        // Update the slider's value label.
+        let numberFormatter: NumberFormatter = {
+            let numberFormatter = NumberFormatter()
+            numberFormatter.maximumFractionDigits = 2
+            return numberFormatter
+        }()
+        valueLabel.text = numberFormatter.string(from: slider.value as NSNumber)
+    }
+    
+    func makeScene(groundOverlay: AGSKMLGroundOverlay) -> AGSScene {
+        // Create a scene for the scene view.
+        let scene = AGSScene(basemap: .imagery())
+        // Set the rotation of the ground overlay.
+        groundOverlay.rotation = -3.046024799346924
+        // Create a KML dataset with the ground overlay as the root node.
+        let dataset = AGSKMLDataset(rootNode: groundOverlay)
+        // Create a KML layer for the scene view.
+        let layer = AGSKMLLayer(kmlDataset: dataset)
+        // Add the layer to the scene.
+        scene.operationalLayers.add(layer)
+        // Move the viewpoint to the ground overlay.
+        let overlayCenter = groundOverlay.geometry.extent.center
+        let camera = AGSCamera(lookAt: overlayCenter, distance: 1250, heading: 45, pitch: 60, roll: 0)
+        let targetExtent = groundOverlay.geometry as! AGSEnvelope
+        sceneView.setViewpoint(AGSViewpoint(targetExtent: targetExtent, camera: camera))
```

Comment:
Since the method still access an instance property `overlay` it is not pure function. Consider moving the set view point portion to scene view `didSet` observer.

Reply:
After moving line 58 (setting rotation), wouldn't it be a pure function since it's not mutating `overlay` or any of its values?

Reply:
Ahh I'm sorry I mean `sceneView` instead of overlay. 😅 
This function access sceneView, which is an instance property. Moving it out can make it a pure function.